// @expo/samples/ExpoConfigView.js
import Constants from 'expo-constants';
import React from 'react';
import { SectionList, Image, StyleSheet, Text, View, FlatList } from 'react-native';

export default class ExpoConfigView extends React.Component {

  constructor(props) {
    super(props)

    this.state = {
      loading: false,
      pet: [],
      url: 'https://pets-unipe.herokuapp.com/pets'
    }
  }

  componentDidMount() {
    this.getPet();
  }

  getPet = () =>{
    this.setState({ loading:true  });
    fetch(this.state.url)
    .then(res => res.json())
    .then(res=>{

        this.setState({
          pet: res,
          loading: false
        })
    })
  };




  render() {
    if(this.state.loading) {
      return (
        <View style={styles.container}>
          <Text style={{fontSize: 36, marginBottom: 25, textAlign: "center"}}>
            Carregando...
          </Text>
        </View>
      );
    }

    return (
      <View style={{flex: 1, paddingTop: 50, paddingLeft:5}}>
        <Text style={{fontSize: 36, marginBottom: 25, textAlign: "center"}}>
            Lista de Pets
        </Text>

          <FlatList

            data={this.state.pet}
            renderItem={
            ({item}) => <Text style={styles.estilizaTexto}> Nome: {item.name} - Raça: {item.raca}</Text>
            }
          />
      </View>
    )

    
  }

 

}



const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
  estilizaTexto: {
    fontSize: 18, 
    backgroundColor:'#E1DFDF', 
    marginTop: 3, marginLeft:32, 
    borderRadius: 2, 
    height: 30, 
    width: 280, 
    textAlign: "center"
  }
});
